plugins/database/cassandra/cassandra.go

```diff
@@ -80,162 +72,230 @@ func (c *Cassandra) getConnection(ctx context.Context) (*gocql.Session, error) {
-	// Grab the lock
+// NewUser generates the username/password on the underlying Cassandra secret backend as instructed by
+// the statements provided.
+func (c *Cassandra) NewUser(ctx context.Context, req newdbplugin.NewUserRequest) (newdbplugin.NewUserResponse, error) {
 	c.Lock()
 	defer c.Unlock()
 
-	statements = dbutil.StatementCompatibilityHelper(statements)
-
-	// Get the connection
 	session, err := c.getConnection(ctx)
 	if err != nil {
-		return "", "", err
+		return newdbplugin.NewUserResponse{}, err
 	}
 
-	creationCQL := statements.Creation
+	creationCQL := req.Statements.Commands
 	if len(creationCQL) == 0 {
 		creationCQL = []string{defaultUserCreationCQL}
 	}
 
-	rollbackCQL := statements.Rollback
+	rollbackCQL := req.RollbackStatements.Commands
 	if len(rollbackCQL) == 0 {
 		rollbackCQL = []string{defaultUserDeletionCQL}
 	}
 
-	username, err = c.GenerateUsername(usernameConfig)
-	username = strings.Replace(username, "-", "_", -1)
+	username, err := credsutil.GenerateUsername(
+		credsutil.DisplayName(req.UsernameConfig.DisplayName, 15),
+		credsutil.RoleName(req.UsernameConfig.RoleName, 15),
+		credsutil.Separator("_"),
+		credsutil.MaxLength(100),
+		credsutil.ToLower(),
+	)
 	if err != nil {
-		return "", "", err
+		return newdbplugin.NewUserResponse{}, err
 	}
-	// Cassandra doesn't like the uppercase usernames
-	username = strings.ToLower(username)
+	username = strings.ReplaceAll(username, "-", "_")
 
-	password, err = c.GeneratePassword()
-	if err != nil {
-		return "", "", err
-	}
-
-	// Execute each query
 	for _, stmt := range creationCQL {
 		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
 			query = strings.TrimSpace(query)
 			if len(query) == 0 {
 				continue
 			}
 
-			err = session.Query(dbutil.QueryHelper(query, map[string]string{
+			m := map[string]string{
 				"username": username,
-				"password": password,
-			})).WithContext(ctx).Exec()
+				"password": req.Password,
+			}
+			err = session.
+				Query(dbutil.QueryHelper(query, m)).
+				WithContext(ctx).
+				Exec()
 			if err != nil {
-				for _, stmt := range rollbackCQL {
-					for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
-						query = strings.TrimSpace(query)
-						if len(query) == 0 {
-							continue
-						}
-
-						session.Query(dbutil.QueryHelper(query, map[string]string{
-							"username": username,
-						})).WithContext(ctx).Exec()
-					}
+				rollbackErr := rollbackUser(ctx, session, username, rollbackCQL)
+				if rollbackErr != nil {
+					err = multierror.Append(err, rollbackErr)
 				}
-				return "", "", err
+				return newdbplugin.NewUserResponse{}, err
 			}
 		}
 	}
 
-	return username, password, nil
+	resp := newdbplugin.NewUserResponse{
+		Username: username,
+	}
+	return resp, nil
+}
+
+func (c *Cassandra) newUsername(config newdbplugin.UsernameMetadata) (string, error) {
```

Comment:
`newUsername`, `trunc` and `joinNonEmpty` should no longer be necessary right?

Reply:
Whoops, you are correct. Removed.